+ source ../common/common_lib.sh
++ source /root/mugen-riscv/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/mugen-riscv/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Thu Dec 29 09:16:59 2022 - INFO  - Start environmental preparation.
+ DNF_INSTALL 'spawn-fcgi php-cgi nginx'
+ pkgs='spawn-fcgi php-cgi nginx'
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py install --pkgs 'spawn-fcgi php-cgi nginx' --node 1 --tempfile ''
+ tmpfile2=/tmp/tmpg5m0rqyf
+ '[' -z '' ']'
+ tmpfile=/tmp/tmpg5m0rqyf
+ echo 'SOCKET=/var/run/fcgiwrap.sock
FCGI_SOCKET=/var/run/fcgiwrap.sock
FCGI_PROGRAM=/usr/bin/php-cgi
FCGI_USER=nginx
FCGI_GROUP=nginx
FCGI_EXTRA_OPTIONS="-M 0777"
OPTIONS="-u $FCGI_USER -g $FCGI_GROUP -s $FCGI_SOCKET -S $FCGI_EXTRA_OPTIONS -F 1 -P /var/run/spawn-fcgi.pid -- $FCGI_PROGRAM"'
+ service=spawn-fcgi.service
++ date '+%Y-%m-%d %T'
+ log_time='2022-12-29 09:18:10'
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Thu Dec 29 09:18:11 2022 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Thu Dec 29 09:18:12 2022 - INFO  - Start testing...
+ test_restart spawn-fcgi.service
+ service=spawn-fcgi.service
+ systemctl restart spawn-fcgi.service
Job for spawn-fcgi.service failed because the control process exited with error code.
See "systemctl status spawn-fcgi.service" and "journalctl -xeu spawn-fcgi.service" for details.
+ CHECK_RESULT 1 0 0 'spawn-fcgi.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'spawn-fcgi.service restart failed'
+ LOG_ERROR 'spawn-fcgi.service restart failed'
+ message='spawn-fcgi.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'spawn-fcgi.service restart failed'
Thu Dec 29 09:18:13 2022 - ERROR - spawn-fcgi.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 34'
+ message='../common/common_lib.sh line 34'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 34'
Thu Dec 29 09:18:13 2022 - ERROR - ../common/common_lib.sh line 34
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status spawn-fcgi.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'spawn-fcgi.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'spawn-fcgi.service restart failed'
+ LOG_ERROR 'spawn-fcgi.service restart failed'
+ message='spawn-fcgi.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'spawn-fcgi.service restart failed'
Thu Dec 29 09:18:20 2022 - ERROR - spawn-fcgi.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 37'
+ message='../common/common_lib.sh line 37'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 37'
Thu Dec 29 09:18:20 2022 - ERROR - ../common/common_lib.sh line 37
+ return 0
+ systemctl stop spawn-fcgi.service
+ CHECK_RESULT 0 0 0 'spawn-fcgi.service stop failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service stop failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status spawn-fcgi.service
+ grep 'Active: inactive'
+ CHECK_RESULT 1 0 0 'spawn-fcgi.service stop failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service stop failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'spawn-fcgi.service stop failed'
+ LOG_ERROR 'spawn-fcgi.service stop failed'
+ message='spawn-fcgi.service stop failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'spawn-fcgi.service stop failed'
Thu Dec 29 09:18:27 2022 - ERROR - spawn-fcgi.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 42'
+ message='../common/common_lib.sh line 42'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 42'
Thu Dec 29 09:18:28 2022 - ERROR - ../common/common_lib.sh line 42
+ return 0
+ systemctl start spawn-fcgi.service
Job for spawn-fcgi.service failed because the control process exited with error code.
See "systemctl status spawn-fcgi.service" and "journalctl -xeu spawn-fcgi.service" for details.
+ CHECK_RESULT 1 0 0 'spawn-fcgi.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'spawn-fcgi.service start failed'
+ LOG_ERROR 'spawn-fcgi.service start failed'
+ message='spawn-fcgi.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'spawn-fcgi.service start failed'
Thu Dec 29 09:18:28 2022 - ERROR - spawn-fcgi.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 44'
+ message='../common/common_lib.sh line 44'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 44'
Thu Dec 29 09:18:29 2022 - ERROR - ../common/common_lib.sh line 44
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status spawn-fcgi.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'spawn-fcgi.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'spawn-fcgi.service start failed'
+ LOG_ERROR 'spawn-fcgi.service start failed'
+ message='spawn-fcgi.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'spawn-fcgi.service start failed'
Thu Dec 29 09:18:35 2022 - ERROR - spawn-fcgi.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 47'
+ message='../common/common_lib.sh line 47'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 47'
Thu Dec 29 09:18:36 2022 - ERROR - ../common/common_lib.sh line 47
+ return 0
+ systemctl enable spawn-fcgi.service
+ grep 'spawn-fcgi.service is not a native service'
spawn-fcgi.service is not a native service, redirecting to systemd-sysv-install.
+ CHECK_RESULT 0 0 0 'spawn-fcgi.service enable failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service enable failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl disable spawn-fcgi.service
+ grep 'spawn-fcgi.service is not a native service'
spawn-fcgi.service is not a native service, redirecting to systemd-sysv-install.
+ CHECK_RESULT 0 0 0 'spawn-fcgi.service disable failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service disable failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ journalctl --since '2022-12-29 09:18:10' -u spawn-fcgi.service
+ grep -i 'fail\|error'
+ grep -v -i 'DEBUG\|INFO\|WARNING'
Dec 29 09:18:12 openEuler-riscv64 systemd[1]: spawn-fcgi.service: Control process exited, code=exited, status=1/FAILURE
Dec 29 09:18:12 openEuler-riscv64 systemd[1]: spawn-fcgi.service: Failed with result 'exit-code'.
Dec 29 09:18:12 openEuler-riscv64 systemd[1]: Failed to start LSB: Start and stop FastCGI processes.
Dec 29 09:18:28 openEuler-riscv64 systemd[1]: spawn-fcgi.service: Control process exited, code=exited, status=1/FAILURE
Dec 29 09:18:28 openEuler-riscv64 systemd[1]: spawn-fcgi.service: Failed with result 'exit-code'.
Dec 29 09:18:28 openEuler-riscv64 systemd[1]: Failed to start LSB: Start and stop FastCGI processes.
+ CHECK_RESULT 0 0 1 'There is an error message for the log of spawn-fcgi.service'
+ actual_result=0
+ expect_result=0
+ mode=1
+ error_log='There is an error message for the log of spawn-fcgi.service'
+ '[' -z 0 ']'
+ '[' 1 -eq 0 ']'
+ test 0x == 0x
+ test -n 'There is an error message for the log of spawn-fcgi.service'
+ LOG_ERROR 'There is an error message for the log of spawn-fcgi.service'
+ message='There is an error message for the log of spawn-fcgi.service'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'There is an error message for the log of spawn-fcgi.service'
Thu Dec 29 09:18:52 2022 - ERROR - There is an error message for the log of spawn-fcgi.service
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_spawn-fcgi.sh line 45'
+ message='oe_test_service_spawn-fcgi.sh line 45'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_spawn-fcgi.sh line 45'
Thu Dec 29 09:18:52 2022 - ERROR - oe_test_service_spawn-fcgi.sh line 45
+ return 0
++ pgrep -f php-cgi
+ first_pid=
+ systemctl reload spawn-fcgi.service
spawn-fcgi.service is not active, cannot reload.
++ pgrep -f php-cgi
+ test -ne ''
oe_test_service_spawn-fcgi.sh: line 48: test: -ne: unary operator expected
+ CHECK_RESULT 2 0 0 'spawn-fcgi.service reload failed'
+ actual_result=2
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service reload failed'
+ '[' -z 2 ']'
+ '[' 0 -eq 0 ']'
+ test 2x '!=' 0x
+ test -n 'spawn-fcgi.service reload failed'
+ LOG_ERROR 'spawn-fcgi.service reload failed'
+ message='spawn-fcgi.service reload failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'spawn-fcgi.service reload failed'
Thu Dec 29 09:18:53 2022 - ERROR - spawn-fcgi.service reload failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_spawn-fcgi.sh line 49'
+ message='oe_test_service_spawn-fcgi.sh line 49'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_spawn-fcgi.sh line 49'
Thu Dec 29 09:18:54 2022 - ERROR - oe_test_service_spawn-fcgi.sh line 49
+ return 0
+ systemctl status spawn-fcgi.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'spawn-fcgi.service reload causes the service status to change'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='spawn-fcgi.service reload causes the service status to change'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'spawn-fcgi.service reload causes the service status to change'
+ LOG_ERROR 'spawn-fcgi.service reload causes the service status to change'
+ message='spawn-fcgi.service reload causes the service status to change'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'spawn-fcgi.service reload causes the service status to change'
Thu Dec 29 09:18:55 2022 - ERROR - spawn-fcgi.service reload causes the service status to change
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_service_spawn-fcgi.sh line 51'
+ message='oe_test_service_spawn-fcgi.sh line 51'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'oe_test_service_spawn-fcgi.sh line 51'
Thu Dec 29 09:18:55 2022 - ERROR - oe_test_service_spawn-fcgi.sh line 51
+ return 0
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Thu Dec 29 09:18:56 2022 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 8
+ test 8 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Thu Dec 29 09:18:56 2022 - ERROR - failed to execute the case.
+ exit 8
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Thu Dec 29 09:18:57 2022 - INFO  - start environment cleanup.
+ systemctl stop spawn-fcgi.service
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmpg5m0rqyf ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmpg5m0rqyf
+ '[' 0 -ne 0 ']'
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Thu Dec 29 09:19:21 2022 - INFO  - Finish environment cleanup!
